### PR TITLE
Add AWS EBS CSI driver operator starter

### PR DESCRIPTION
**Pull request by jsafrane**
Implement AWS EBS CSI driver operator.

* Add (a lot of) clients into `Clients` structure in `pkg/clients`, so we don't needs to pass individual clients around.
* Add generic CSI driver operator code into `pkg/operator/`
  * Notice `OperatorConfig` and `OperatorControllerConfig` in `pkg/operator/config` - those will configure `starter.go`.
    * Both configure starter.go, `OperatorConfig` is needed to create `Clients` and `OperatorControllerConfig` needs `Clients` to already exist, to instantiate informers for the operator. Feedback is welcome.
  * `pkg/operator/starter.go` should be generic enough for all CSI driver operators. Maybe it is not now, but we'll extend it as the need arises (IMO Azure should be just fine).

* Add `cmd/aws-ebs-csi-driver-operator/main.go` and update `assets/overlays/aws-ebs/aws_ebs.go` that populates `OperatorConfig` and `OperatorControllerConfig` with AWS specific configuration and run (hopefully generic) starter.go. All hooks are copy-paste from the existing operator. Question is, if I forgot any of them.

Vendor is in a separate commit, if it helps with github review.

### CODE REVIEW

**Comment by gnufied** [comment]
Unrelated to this PR nit: We seem to be embedding all files in assets, including go file. Seems unnecessary.

File: assets/overlays/aws-ebs/aws_ebs.go [hunk 1]
```diff
@@ -70,3 +104,315 @@ func GetAWSEBSGeneratorConfig() *generator.CSIDriverGeneratorConfig {
...
+		// Standalone-only hooks
+		cfg.AddDeploymentHookBuilders(c, getCustomAWSCABundleBuilder(cloudConfigName))
+	} else {
+		// HyperShift only hooks
+		cfg.AddDeploymentHookBuilders(c, getCustomAWSCABundleBuilder("user-ca-bundle"))
+	}
+	cfg.DeploymentWatchedSecretNames = append(cfg.DeploymentWatchedSecretNames, cloudCredSecretName, metricsCertSecretName)
+
+	// Controllers to run on all clusters
+	ctrl, err := newCustomAWSBundleSyncer(c)
```
Comment (gnufied) [inline assets/overlays/aws-ebs/aws_ebs.go:140]: Does this needs to happen in hypershift clusters too?

**Comment by gnufied** [comment]
nit: The yaml comments:

```
# Applied strategic merge patch provisioner.yaml
```

Tells us that a strategic merge patch was applied to `provisioner.yaml` but it does not tells us, what exactly was the patch. I guess it is getting printed because of a previously applied patch and in which case I am not sure if it is useful.

**Comment by gnufied** [comment]
Overall obviously, using `assets` name for directory that contains entirety of driver code seems strange. One idea I had was - to rename `assets` to `drivers` and still keep same everything.  @jsafrane let me know what you think.

File: assets/overlays/aws-ebs/aws_ebs.go [hunk 2]
```diff
@@ -70,3 +104,315 @@ func GetAWSEBSGeneratorConfig() *generator.CSIDriverGeneratorConfig {
 		},
 	}
 }
+
+func GetAWSEBSOperatorConfig() *config.OperatorConfig {
```
[the remark this answers is not in the document]
Comment (jsafrane) [reply assets/overlays/aws-ebs/aws_ebs.go:108]: Added some comments.

File: pkg/operator/config/defaults/hooks.go [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+var (
+	hostedControlPlaneGVR = schema.GroupVersionResource{
+		Group:    "hypershift.openshift.io",
+		Version:  "v1beta1",
+		Resource: "hostedcontrolplanes",
+	}
+)
+
+// NewDefaultOperatorConfig returns an OperatorConfig with default standalone / HyperShift hooks.
+func NewDefaultOperatorConfig(flavour generator.ClusterFlavour, c *clients.Clients, controllerNamePrefix string) *config.OperatorControllerConfig {
```
[the remark this answers is not in the document]
Comment (jsafrane) [reply pkg/operator/config/defaults/hooks.go:31]: Renamed

File: assets/overlays/aws-ebs/aws_ebs.go (hunk 1 above)
Comment (jsafrane) [reply assets/overlays/aws-ebs/aws_ebs.go:140]: Good catch, this is standalone only. Fixed

**Comment by jsafrane** [comment]
> nit: The yaml comments:
> ```
> # Applied strategic merge patch provisioner.yaml
> ```
> Tells us that a strategic merge patch was applied to provisioner.yaml but it does not tells us, what exactly was the patch. I guess it is getting printed because of a previously applied patch and in which case I am not sure if it is useful.

It's just above the line you copied

```
# provisioner.yaml: Loaded from common/sidecars/provisioner.yaml
# provisioner.yaml: Added arguments [--default-fstype=ext4 --feature-gates=Topology=true --extra-create-metadata=true --timeout=60s]
# Applied strategic merge patch provisioner.yaml

```

**Pull request merged**